diesel/src/types/mod.rs

```diff
@@ -274,10 +275,10 @@ pub type VarChar = Text;
 pub use pg::types::sql_types::*;
 
 pub trait HasSqlType<ST>: TypeMetadata {
-    fn metadata() -> Self::TypeMetadata;
+    fn metadata(lookup: &Self::MetadataLookup) -> Self::TypeMetadata;
```

Comment:
The `PgTypeMetadata::Dynamic`/`PgTypeMetadata::Static` approach does not require to change this for all backends. So this may be preferred? Or is this required for any future feature on other backends?